core/dbt/cli/options.py

```diff
@@ -42,3 +59,17 @@ def parser_process(value, state):
                 our_parser.process = parser_process
                 break
         return retval
+
+    def type_cast_value(self, ctx: Context, value: t.Any) -> t.Any:
+        def flatten(data):
+            if isinstance(data, tuple):
+                for x in data:
+                    yield from flatten(x)
+            else:
+                yield data
+
+        # there will be nested tuples to flatten when multiple=True
+        value = super(MultiOption, self).type_cast_value(ctx, value)
+        if value:
+            value = tuple(flatten(value))
```

Comment:
I don't know why, but the value can be `None`. We only want to flatten it when it has a non-None value.